lib/formatter/phone_number.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'pry'
+
+module Formatter
+  class PhoneNumber
+    InvalidNumber = 'Error!'
+    
+    class << self
```

Comment:
> this is the first time i’ve worked with class methods and i don’t get why i have to declare them as such on line 9

Interesting, I deliberately chose to use class methods as I feel it can be a thing people might not have encountered but are very common. 

doing this `class << self` and then defining methods inside is the same as doing 

``` 
def self.call(number)
  ...
end 

private 

def self.remove_whitespace(number)
....
end
```

Using class methods calls the class method on the PhoneNumber class, not the instance method on the instance of the class. 

So if we created an instance of the class: 
```
formatter = Formatter::PhoneNumber.new
formatter.call # calling an instance method 
```

But if we want to call the class method we would invoke it like this:

```
PhoneNumber.call
```

Ruby needs to know if you're defining a class method or an instance method (so that it can route the `.call` to the right method) and that's why we have these 2 ways to declare methods. 

If we don't need the object to hold state then it's effectively a functional method that always returns the same when we give it the same value (e.g. whenever we give it `0741 2525 804` it will ALWAYS return `07412525804`). In this case we don't need to create an object, we can use class methods. This is more efficient on memory (because we don't have to store the instance of the object in memory), but also makes it clearer to developers that it's a functional method and isn't intended to store state. 

Let me know if anything above doesn't make sense or if you have any other questions off the back of it!